### PR TITLE
[BEAM-59] Some low hanging fruit in FileSystems/LocalFileSystem/LocalResource

### PR DESCRIPTION
**Pull request by dhalperi**
Pulling out from ongoing work to convert `FileBasedSink`.

R: @ssisk or @peihe or @reuvenlax as available

### CODE REVIEW

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalResourceIdTest.java [hunk 1]
```diff
@@ -214,6 +223,20 @@ public void testEquals() throws Exception {
         toResourceIdentifier("/root/tmp/"));
   }
 
+  @Test
+  public void testToString() throws Exception {
+    File someFile = tmpFolder.newFile("somefile");
+    LocalResourceId fileResource = LocalResourceId.fromPath(someFile.toPath(), false);
```
Comment (peihe) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalResourceIdTest.java]: false /* isDirectory */

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalResourceIdTest.java [hunk 2]
```diff
@@ -214,6 +223,20 @@ public void testEquals() throws Exception {
...
+    assertThat(fileResource.toString(), not(endsWith(File.separator)));
+    assertThat(fileResource.toString(), containsString("somefile"));
+    assertThat(fileResource.toString(), startsWith(tmpFolder.getRoot().getAbsolutePath()));
+
+    LocalResourceId dirResource = LocalResourceId.fromPath(someFile.toPath(), true);
```
Comment (peihe) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalResourceIdTest.java]: true /* isDirectory */

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalFileSystemTest.java [hunk 3]
```diff
@@ -121,36 +122,68 @@ public void testReadNonExistentFile() throws Exception {
...
+    createFileWithContent(srcPath1, "content1");
+    createFileWithContent(srcPath2, "content2");
+
+    localFileSystem.rename(
+        toLocalResourceIds(ImmutableList.of(srcPath1, srcPath2), false /* isDirectory */),
+        toLocalResourceIds(ImmutableList.of(destPath1, destPath2), false /* isDirectory */));
+
+    assertContents(
+        ImmutableList.of(destPath1, destPath2),
+        ImmutableList.of("content1", "content2"));
```
Comment (peihe) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalFileSystemTest.java]: verify srcPath1 and srcPath2 don't exist.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalFileSystemTest.java [hunk 4]
```diff
@@ -121,36 +122,68 @@ public void testReadNonExistentFile() throws Exception {
...
+
   @Test
   public void testCopyWithExistingSrcFile() throws Exception {
     Path srcPath1 = temporaryFolder.newFile().toPath();
     Path srcPath2 = temporaryFolder.newFile().toPath();
 
-    Path destPath1 = srcPath1.resolveSibling("dest1");
+    Path destPath1 =
+        temporaryFolder.getRoot().toPath().resolve("nonexistentdir").resolve("dest1");
     Path destPath2 = srcPath2.resolveSibling("dest2");
```
Comment (peihe) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalFileSystemTest.java]: remove extra line

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalResourceIdTest.java (hunk 1 above)
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalResourceIdTest.java]: done

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalResourceIdTest.java (hunk 2 above)
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalResourceIdTest.java]: done

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalFileSystemTest.java (hunk 3 above)
Comment (dhalperi) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalFileSystemTest.java]: done

**Pull request closed without merging**